src/components/Editor/Breakpoint.js

```diff
@@ -48,6 +53,10 @@ class Breakpoint extends Component {
     const sourceId = selectedSource.get("id");
     const line = toEditorLine(sourceId, breakpoint.location.line);
 
+    if (isEmptyLine(editor.codeMirror, line)) {
```

Comment:
@darkwing this might help clarify @jasonLaster comments. Whereby the reducer has all the data about the empty lines so it should be able to let us know what an empty line is.

Hope this helps!

``` diff
diff --git a/src/components/Editor/Breakpoint.js b/src/components/Editor/Breakpoint.js
index 767b598b..386a69ea 100644
--- a/src/components/Editor/Breakpoint.js
+++ b/src/components/Editor/Breakpoint.js
@@ -6,12 +6,7 @@ import ReactDOM from "react-dom";
 import classnames from "classnames";
 import Svg from "../shared/Svg";

-import {
-  getDocument,
-  showSourceText,
-  toEditorLine,
-  isEmptyLine
-} from "../../utils/editor";
+import { getDocument, showSourceText, toEditorLine } from "../../utils/editor";

 const breakpointSvg = document.createElement("div");
 ReactDOM.render(Svg("breakpoint"), breakpointSvg);
@@ -31,7 +26,8 @@ class Breakpoint extends Component {
     breakpoint: Object,
     selectedSource: Object,
     line: number,
-    editor: Object
+    editor: Object,
+    isEmptyLine: Function
   };

   addBreakpoint: Function;
@@ -42,7 +38,7 @@ class Breakpoint extends Component {
   }

   addBreakpoint() {
-    const { breakpoint, editor, selectedSource } = this.props;
+    const { breakpoint, editor, selectedSource, isEmptyLine } = this.props;

     // NOTE: we need to wait for the breakpoint to be loaded
     // to get the generated location
diff --git a/src/components/Editor/Breakpoints.js b/src/components/Editor/Breakpoints.js
index a51252d0..194028c2 100644
--- a/src/components/Editor/Breakpoints.js
+++ b/src/components/Editor/Breakpoints.js
@@ -5,7 +5,7 @@ import React, { Component } from "react";
 import Breakpoint from "./Breakpoint";

 import actions from "../../actions";
-import { getSelectedSource } from "../../selectors";
+import { getSelectedSource, isEmptyLineInSource } from "../../selectors";
 import getVisibleBreakpoints from "../../selectors/visibleBreakpoints";
 import { makeLocationId } from "../../utils/breakpoint";

@@ -29,7 +29,7 @@ class Breakpoints extends Component {
   }

   render() {
-    const { breakpoints, selectedSource, editor } = this.props;
+    const { breakpoints, selectedSource, editor, isEmptyLine } = this.props;

     if (!selectedSource || !breakpoints || selectedSource.get("isBlackBoxed")) {
       return null;
@@ -44,6 +44,7 @@ class Breakpoints extends Component {
               breakpoint={bp}
               selectedSource={selectedSource}
               editor={editor}
+              isEmptyLine={isEmptyLine}
             />
           );
         })}
@@ -57,7 +58,8 @@ Breakpoints.displayName = "Breakpoints";
 export default connect(
   state => ({
     breakpoints: getVisibleBreakpoints(state),
-    selectedSource: getSelectedSource(state)
+    selectedSource: getSelectedSource(state),
+    isEmptyLine: line => isEmptyLineInSource(state, line)
   }),
   dispatch => bindActionCreators(actions, dispatch)
 )(Breakpoints);
diff --git a/src/components/Editor/index.js b/src/components/Editor/index.js
index 5589daf8..ba946317 100644
--- a/src/components/Editor/index.js
+++ b/src/components/Editor/index.js
@@ -26,7 +26,8 @@ import {
   getFileSearchQueryState,
   getFileSearchModifierState,
   getVisibleBreakpoints,
-  getInScopeLines
+  getInScopeLines,
+  isEmptyLineInSource
 } from "../../selectors";

 import actions from "../../actions";
@@ -55,8 +56,7 @@ import {
   toEditorLine,
   toEditorPosition,
   toEditorRange,
-  resetLineNumberFormat,
-  isEmptyLine
+  resetLineNumberFormat
 } from "../../utils/editor";

 import { isFirefox } from "devtools-config";
@@ -368,7 +368,7 @@ class Editor extends PureComponent {
   }

   onGutterClick(cm, line, gutter, ev) {
-    const { selectedSource, toggleBreakpoint } = this.props;
+    const { selectedSource, toggleBreakpoint, isEmptyLine } = this.props;

     // ignore right clicks in the gutter
     if (
@@ -401,7 +401,8 @@ class Editor extends PureComponent {
       selectedSource,
       breakpoints,
       toggleBreakpoint,
-      toggleDisabledBreakpoint
+      toggleDisabledBreakpoint,
+      isEmptyLine
     } = this.props;

     if (selectedSource && selectedSource.get("isBlackBoxed")) {
@@ -815,7 +816,8 @@ export default connect(
       query: getFileSearchQueryState(state),
       searchModifiers: getFileSearchModifierState(state),
       linesInScope: getInScopeLines(state),
-      selection: getSelection(state)
+      selection: getSelection(state),
+      isEmptyLine: line => isEmptyLineInSource(state, line)
     };
   },
   dispatch => bindActionCreators(actions, dispatch)
diff --git a/src/reducers/ast.js b/src/reducers/ast.js
index d5fba776..1e3bb8af 100644
--- a/src/reducers/ast.js
+++ b/src/reducers/ast.js
@@ -132,6 +132,11 @@ export function hasSymbols(state: OuterState, source: Source): boolean {
   return !!state.ast.getIn(["symbols", source.id]);
 }

+export function isEmptyLineInSource(state: OuterState, line) {
+  const emptyLines = getEmptyLines(state, selectedSource);
+  return emptyLines.includes(line);
+}
+
 export function getEmptyLines(state: OuterState, source: Source) {
   if (!source) {
     return false;
(END)
```